set.mm

```diff
@@ -69732,6 +69740,24 @@ empty set when it is not meaningful (as shown by ~ ndmfv and ~ fvprc ).
       AHBKQLAGMFAGBCNOP $.
   $}
 
+  ${
+    $d M y $.  $d N x y $.  $d X x y $.  $d V x $.  $d ps x $.
+    fvmptrab.f $e |- F = ( x e. V |-> { y e. M | ph } ) $.
+    fvmptrab.r $e |- ( x = X -> ( ph <-> ps ) ) $.
+    fvmptrab.s $e |- ( x = X -> M = N ) $.
+    fvmptrab.v $e |- ( X e. V -> N e. _V ) $.
+    fvmptrab.n $e |- ( X e/ V -> N = (/) ) $.
+    $( Implications for the value of a function defined by the maps-to notation
+       with a class abstraction as a result having an element.  (Contributed by
+       Alexander van der Vekens, 15-Jul-2018.) $)
+    fvmptrab $p |- ( F ` X ) = { y e. N | ps } $=
```

Comment:
I'm not sure it is worth adding this to Main. It relies on the fact that out-of-domain arguments evaluate to the empty set, which relies on set.mm's particular encoding. It looks like it has a single usage, http://us2.metamath.org/mpeuni/uvtxaval.html, where it would shorten steps around http://us2.metamath.org/mpeuni/fvmptd.html ? To me, the current proof looks preferable.

Reply:
I think this pattern is used very often in set.mm, especially to avoid the distiction between sets and proper classes. In the case of uvtxval, the left hand side as well as the right hand side relies on this specialty ( { v e. V | .. } = { v e. ( Vtx `G ) | .. } = { v e. (/) | .. } = (/) if -. G e. _V). Would it be OK if I revise fvmptrab so that N is replaced by a function value ( G ` X ) ?

Reply:
Do you mean something like
```
$e ...
$e |- ( X e. dom F <-> Y e. dom G )
$p |- ( F ` X ) = { y e. ( G ` Y ) | ps }
```
?

Reply:
Your new statement does not depend on the encoding and looks more natural, so I prefer it and I think it's ok to have it in Main (although I do not know if it is widely applicable).  The comment has problems ("result having" and "an function"). Maybe:
> Value of a function defined in maps-to notation with value a particular kind of restricted abstraction (the function ` F ` is the subfunction of ` G ` cut out by ` ph ` ).

Does it hold when _V is replaced by V and you add $p |- X e. V -> ( ... ? (or use deduction form)

Reply:
And do you think that the extra generality of my above proposal with `( X e. dom F <-> Y e. dom G )` is useless ? (it may very well be)